src/error.rs

```diff
@@ -19,3 +19,11 @@ impl From<std::io::Error> for Error {
         Error::new(&e.to_string())
     }
 }
+
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&self.details)
+    }
+}
+
+impl std::error::Error for Error {}
```

Comment:
It's best if custom error types implement the std error trait. This allows using crates like `anyhow` or simply `Box<dyn std::error::Error>` without extra boilerplate.

Reply:
I'll add more errors as in my other projects, generated by a make_error macro that looks pretty much exactly like this.

Reply:
In case you don't know about it, the [thiserror](https://crates.io/crates/thiserror/1.0.24) crate looks similar to what you're describing there